configure.ac

```diff
@@ -1410,7 +1410,7 @@ case "$host_cpu-$host_vendor-$host_os" in
     ##
     ## POSIX feature information can be found in the gcc manual at:
     ## http://www.gnu.org/s/libc/manual/html_node/Feature-Test-Macros.html
-    H5_CPPFLAGS="-D_POSIX_C_SOURCE=200809L $H5_CPPFLAGS"
+    H5_CPPFLAGS="-D_BSD_SOURCE -D_POSIX_C_SOURCE=200809L $H5_CPPFLAGS"
```

Comment:
Did you test this? The change here is being applied to Linux.

Reply:
I tested it on OpenBSD, but not other OSes.  The CI here on github seems to be green.  But as I said, I'm not sure if having this defined could be detrimental on other OSes or not...

Reply:
> I tested it on OpenBSD, but not other OSes. The CI here on github seems to be green. But as I said, I'm not sure if having this defined could be detrimental on other OSes or not...

I think this was removed because it was flagged as a warning in newer gcc builds? Anyway it should probably be just defined for openBSD.

Reply:
I wouldn't be surprised if it was also needed for the other BSDs... do you do any testing with them?  I could also try on FreeBSD...

Reply:
So gcc 12 outputs the following for this:
# warning "_BSD_SOURCE  is deprecated, use _DEFAULT_SOURCE"

Reply:
Some quick googling suggests this warning is from glibc.

Alas, using `_DEFAULT_SOURCE` instead of `_BSD_SOURCE` does not work for OpenBSD:

```
builder3$ cc -fsyntax-only -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200809L test.c
test.c:5:3: warning: implicit declaration of function 'flock' is invalid in C99 [-Wimplicit-function-declaration]
  flock(0,0);
  ^
```

From what I can tell, `flock` is *not* part of POSIX, but rather is from BSD. See: https://gavv.github.io/articles/file-locks/#bsd-locks-flock

I suspect OpenBSD interprets `_POSIX_C_SOURCE=200809L` to mean to include *only* things in POSIX, and thus `flock` becomes hidden.  OTOH, glibc seems to interpret `_POSIX_C_SOURCE=200809L` differently.

Reply:
Including -D_BSD_SOURCE on OpenBSD is fine, I'd just like to limit it to that OS. Also, we shouldn't be applying that flag to any untested operating systems. i.e., unless we've tested on NetBSD or FreeBSD, we shouldn't be blindly applying the flag there.

Reply:
I don't know much about cmake/autotools, how can this be done?

Reply:
CMake has this line in CMakeDetermineSystem.cmake:
    elseif(CMAKE_HOST_SYSTEM_NAME MATCHES "OpenBSD")

I think it sets this from a uname call so, CMake could check this var and add the the define.

Reply:
I think autotools uses a "$host_os" var - at least in setting up flags. I think this var is set in configure.ac.

Reply:
OK, please see my new change.